api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -65,17 +60,12 @@ func (NutanixPrismCentralEndpointSpec) VariableSchema() clusterv1.VariableSchema
 			Description: "Nutanix Prism Central endpoint configuration",
 			Type:        "object",
 			Properties: map[string]clusterv1.JSONSchemaProps{
-				"host": {
-					Description: "the DNS name or IP address of the Nutanix Prism Central",
+				"url": {
+					Description: "The URL of Nutanix Prism Central, can be DNS name or an IP address",
 					Type:        "string",
 					MinLength:   ptr.To[int64](1),
-				},
-				"port": {
-					Description: "The port number to access the Nutanix Prism Central",
-					Type:        "integer",
-					Default:     variables.MustMarshal(PrismCentralPort),
-					Minimum:     ptr.To[int64](1),
-					Maximum:     ptr.To[int64](65535),
+					Format:      "uri",
+					Pattern:     "^https://",
```

Comment:
are we allowing `http://` with `insecure: true`

Reply:
No, the schema is always `https` (the `Pattern` enforces that). `insecure: true` is only used for `TLSSkipVerify` when the serving cert was signed by a self-signed CA.

Reply:
Totally forgot we had this package, thanks @faiq 
Fixed in https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/36/commits/dd1af816d19d6bdc1eacffed45291002aad7db7c

Reply:
nice.